ext/openssl/openssl.c

```diff
@@ -703,7 +706,13 @@ static void add_assoc_name_entry(zval * val, char * key, X509_NAME * name, int s
 				add_assoc_stringl(&subitem, sname, (char *)to_add, to_add_len);
 			}
 		}
+
+		if (needs_free) {
+			/* ASN1_STRING_to_UTF8(3): The buffer out should be freed using free(3) */
+			free(to_add);
```

Comment:
Yep it was probably bug in the docs that got fixed as the implementation (even for 0.9.8) uses OPENSSL_malloc.